### PR TITLE
rt-fs: Add cat command

### PR DESCRIPTION
**Pull request by yahavi**
Add `cat` command:
```
rt-fs cat <path>
```

### CODE REVIEW

File: rt-fs/commands/cat.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	"github.com/jfrog/jfrog-cli-core/artifactory/commands/generic"
+	"github.com/jfrog/jfrog-cli-core/artifactory/spec"
+	"github.com/jfrog/jfrog-cli-core/artifactory/utils"
+	clientutils "github.com/jfrog/jfrog-client-go/artifactory/services/utils"
+
+	"github.com/jfrog/jfrog-cli-core/plugins/components"
+)
+
+const (
+	downloadThreads      = 3
```
Comment (eyalbe4) [inline rt-fs/commands/cat.go:20]: Since we're always downloading only 1 file, maybe we should change the value to 1.

File: rt-fs/commands/cat.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+	if len(splitRes) > 2 {
+		path = downloadPath[len(repo)+1 : len(downloadPath)-len(name)-1]
+	} else {
+		path = "."
+	}
+
+	return clientutils.Aql{
+		ItemsFind: fmt.Sprintf(`{"repo":"%s","path":"%s","name":"%s"}`, repo, path, name),
+	}
+
```
Comment (eyalbe4) [inline rt-fs/commands/cat.go:120]: Redundant new-line.

File: rt-fs/main.go [hunk 3]
```diff
@@ -21,5 +21,7 @@ func getApp() components.App {
```
Comment (eyalbe4) [inline rt-fs/main.go:21]: Let's change the version from v1.0.0 to v1.1.0
Let's also update the release notes.

File: rt-fs/README.md [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # rt-fs
 
 ## About this plugin
-This plugin display files of a given folder in Artifactory. It designed to mimic the functionality of the Linux/Unix 'ls' command.
+This plugin executes file system commands in Artifactory. It designed to mimic the functionality of the Linux/Unix 'ls' and 'cat' commands.
```
Comment (eyalbe4) [inline rt-fs/README.md:4]: ...It designed... --> ...It is designed...

File: rt-fs/README.md [hunk 5]
```diff
@@ -25,9 +25,19 @@ Uninstalling a plugin
         - server-id: Artifactory server ID configured using the config command **[Optional]**
```
Comment (eyalbe4) [inline rt-fs/README.md:25]: server-id --> --server-id

File: rt-fs/README.md [hunk 6]
```diff
@@ -25,9 +25,19 @@ Uninstalling a plugin
...
-  file_name1.zip   file_name2.zip   file_name3.zip
-  ```
+        $ jfrog rt-fs ls generic-local
+        file_name1.zip   file_name2.zip   file_name3.zip
+    ```
+* cat
+    - Arguments:
+        - path - Path in Artifactory.
+    - Flags:
+        - server-id: Artifactory server ID configured using the config command **[Optional]**
```
Comment (eyalbe4) [inline rt-fs/README.md:35]: server-id --> --server-id

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
One of the tests seems to be failing for me:

```
$ go test -v ./...
?   	github.com/jfrog/jfrog-cli-plugins/rt-fs	[no test files]
=== RUN   TestCreateAql
=== RUN   TestCreateAql/{repository/filename_{"repo":"repository","path":".","name":"filename"}}
=== RUN   TestCreateAql/{repository/dir/filename_{"repo":"repository","path":"dir","name":"filename"}}
=== RUN   TestCreateAql/{repository/dir1/dir2/filename_{"repo":"repository","path":"dir1/dir2","name":"filename"}}
--- PASS: TestCreateAql (0.00s)
    --- PASS: TestCreateAql/{repository/filename_{"repo":"repository","path":".","name":"filename"}} (0.00s)
    --- PASS: TestCreateAql/{repository/dir/filename_{"repo":"repository","path":"dir","name":"filename"}} (0.00s)
    --- PASS: TestCreateAql/{repository/dir1/dir2/filename_{"repo":"repository","path":"dir1/dir2","name":"filename"}} (0.00s)
=== RUN   TestTrimPrefixFromPath
=== RUN   TestTrimPrefixFromPath/{abc_abc_abc}
=== RUN   TestTrimPrefixFromPath/{abc_abc/def_def}
=== RUN   TestTrimPrefixFromPath/{abc/_abc/def_def}
=== RUN   TestTrimPrefixFromPath/{abc/def_abc/def_def}
=== RUN   TestTrimPrefixFromPath/{abc/def/_abc/def_def}
--- PASS: TestTrimPrefixFromPath (0.00s)
    --- PASS: TestTrimPrefixFromPath/{abc_abc_abc} (0.00s)
    --- PASS: TestTrimPrefixFromPath/{abc_abc/def_def} (0.00s)
    --- PASS: TestTrimPrefixFromPath/{abc/_abc/def_def} (0.00s)
    --- PASS: TestTrimPrefixFromPath/{abc/def_abc/def_def} (0.00s)
    --- PASS: TestTrimPrefixFromPath/{abc/def/_abc/def_def} (0.00s)
=== RUN   TestProcessSearchResults
=== RUN   TestProcessSearchResults/{sample1.json_libs-release-local_[org_.nupkg]_[folder_file]_6}
=== RUN   TestProcessSearchResults/{sample1.json_libs-release-local/_[org_.nupkg]_[folder_file]_6}
=== RUN   TestProcessSearchResults/{sample2.json_libs-release-local/org/jfrog/example/gradle/_[services_shared_api_webservice_gradle-example-ci-server]_[folder_folder_folder_folder_folder_folder]_24}
=== RUN   TestProcessSearchResults/{sample3.json_libs-release-local/org_[org]_[folder_file]_3}
--- PASS: TestProcessSearchResults (0.00s)
    --- PASS: TestProcessSearchResults/{sample1.json_libs-release-local_[org_.nupkg]_[folder_file]_6} (0.00s)
    --- PASS: TestProcessSearchResults/{sample1.json_libs-release-local/_[org_.nupkg]_[folder_file]_6} (0.00s)
    --- PASS: TestProcessSearchResults/{sample2.json_libs-release-local/org/jfrog/example/gradle/_[services_shared_api_webservice_gradle-example-ci-server]_[folder_folder_folder_folder_folder_folder]_24} (0.00s)
    --- PASS: TestProcessSearchResults/{sample3.json_libs-release-local/org_[org]_[folder_file]_3} (0.00s)
=== RUN   TestProcessSearchResultsNotExist
    ls_test.go:71: 
        	Error Trace:	ls_test.go:71
        	Error:      	An error is expected but got nil.
        	Test:       	TestProcessSearchResultsNotExist
--- FAIL: TestProcessSearchResultsNotExist (0.00s)
=== RUN   TestShouldRunSecondSearch
=== RUN   TestShouldRunSecondSearch/{sample1.json_libs-release-local_false}
=== RUN   TestShouldRunSecondSearch/{sample1.json_libs-release-local/_false}
=== RUN   TestShouldRunSecondSearch/{sample2.json_libs-release-local/org/jfrog/example/gradle/_false}
=== RUN   TestShouldRunSecondSearch/{sample2.json_libs-release-local/org/_false}
=== RUN   TestShouldRunSecondSearch/{sample3.json_libs-release-local/org_true}
--- PASS: TestShouldRunSecondSearch (0.00s)
    --- PASS: TestShouldRunSecondSearch/{sample1.json_libs-release-local_false} (0.00s)
    --- PASS: TestShouldRunSecondSearch/{sample1.json_libs-release-local/_false} (0.00s)
    --- PASS: TestShouldRunSecondSearch/{sample2.json_libs-release-local/org/jfrog/example/gradle/_false} (0.00s)
    --- PASS: TestShouldRunSecondSearch/{sample2.json_libs-release-local/org/_false} (0.00s)
    --- PASS: TestShouldRunSecondSearch/{sample3.json_libs-release-local/org_true} (0.00s)
FAIL
FAIL	github.com/jfrog/jfrog-cli-plugins/rt-fs/commands	0.314s
FAIL

```

**Comment by yahavi** [comment]
> One of the tests seems to be failing for me:
> 
> ```
> $ go test -v ./...
> ?   	github.com/jfrog/jfrog-cli-plugins/rt-fs	[no test files]
> === RUN   TestCreateAql
> === RUN   TestCreateAql/{repository/filename_{"repo":"repository","path":".","name":"filename"}}
> === RUN   TestCreateAql/{repository/dir/filename_{"repo":"repository","path":"dir","name":"filename"}}
> === RUN   TestCreateAql/{repository/dir1/dir2/filename_{"repo":"repository","path":"dir1/dir2","name":"filename"}}
> --- PASS: TestCreateAql (0.00s)
>     --- PASS: TestCreateAql/{repository/filename_{"repo":"repository","path":".","name":"filename"}} (0.00s)
>     --- PASS: TestCreateAql/{repository/dir/filename_{"repo":"repository","path":"dir","name":"filename"}} (0.00s)
>     --- PASS: TestCreateAql/{repository/dir1/dir2/filename_{"repo":"repository","path":"dir1/dir2","name":"filename"}} (0.00s)
> === RUN   TestTrimPrefixFromPath
> === RUN   TestTrimPrefixFromPath/{abc_abc_abc}
> === RUN   TestTrimPrefixFromPath/{abc_abc/def_def}
> === RUN   TestTrimPrefixFromPath/{abc/_abc/def_def}
> === RUN   TestTrimPrefixFromPath/{abc/def_abc/def_def}
> === RUN   TestTrimPrefixFromPath/{abc/def/_abc/def_def}
> --- PASS: TestTrimPrefixFromPath (0.00s)
>     --- PASS: TestTrimPrefixFromPath/{abc_abc_abc} (0.00s)
>     --- PASS: TestTrimPrefixFromPath/{abc_abc/def_def} (0.00s)
>     --- PASS: TestTrimPrefixFromPath/{abc/_abc/def_def} (0.00s)
>     --- PASS: TestTrimPrefixFromPath/{abc/def_abc/def_def} (0.00s)
>     --- PASS: TestTrimPrefixFromPath/{abc/def/_abc/def_def} (0.00s)
> === RUN   TestProcessSearchResults
> === RUN   TestProcessSearchResults/{sample1.json_libs-release-local_[org_.nupkg]_[folder_file]_6}
> === RUN   TestProcessSearchResults/{sample1.json_libs-release-local/_[org_.nupkg]_[folder_file]_6}
> === RUN   TestProcessSearchResults/{sample2.json_libs-release-local/org/jfrog/example/gradle/_[services_shared_api_webservice_gradle-example-ci-server]_[folder_folder_folder_folder_folder_folder]_24}
> === RUN   TestProcessSearchResults/{sample3.json_libs-release-local/org_[org]_[folder_file]_3}
> --- PASS: TestProcessSearchResults (0.00s)
>     --- PASS: TestProcessSearchResults/{sample1.json_libs-release-local_[org_.nupkg]_[folder_file]_6} (0.00s)
>     --- PASS: TestProcessSearchResults/{sample1.json_libs-release-local/_[org_.nupkg]_[folder_file]_6} (0.00s)
>     --- PASS: TestProcessSearchResults/{sample2.json_libs-release-local/org/jfrog/example/gradle/_[services_shared_api_webservice_gradle-example-ci-server]_[folder_folder_folder_folder_folder_folder]_24} (0.00s)
>     --- PASS: TestProcessSearchResults/{sample3.json_libs-release-local/org_[org]_[folder_file]_3} (0.00s)
> === RUN   TestProcessSearchResultsNotExist
>     ls_test.go:71: 
>         	Error Trace:	ls_test.go:71
>         	Error:      	An error is expected but got nil.
>         	Test:       	TestProcessSearchResultsNotExist
> --- FAIL: TestProcessSearchResultsNotExist (0.00s)
> === RUN   TestShouldRunSecondSearch
> === RUN   TestShouldRunSecondSearch/{sample1.json_libs-release-local_false}
> === RUN   TestShouldRunSecondSearch/{sample1.json_libs-release-local/_false}
> === RUN   TestShouldRunSecondSearch/{sample2.json_libs-release-local/org/jfrog/example/gradle/_false}
> === RUN   TestShouldRunSecondSearch/{sample2.json_libs-release-local/org/_false}
> === RUN   TestShouldRunSecondSearch/{sample3.json_libs-release-local/org_true}
> --- PASS: TestShouldRunSecondSearch (0.00s)
>     --- PASS: TestShouldRunSecondSearch/{sample1.json_libs-release-local_false} (0.00s)
>     --- PASS: TestShouldRunSecondSearch/{sample1.json_libs-release-local/_false} (0.00s)
>     --- PASS: TestShouldRunSecondSearch/{sample2.json_libs-release-local/org/jfrog/example/gradle/_false} (0.00s)
>     --- PASS: TestShouldRunSecondSearch/{sample2.json_libs-release-local/org/_false} (0.00s)
>     --- PASS: TestShouldRunSecondSearch/{sample3.json_libs-release-local/org_true} (0.00s)
> FAIL
> FAIL	github.com/jfrog/jfrog-cli-plugins/rt-fs/commands	0.314s
> FAIL
> ```

Thanks. Fixed.

**Pull request merged**